### PR TITLE
Add `get_last_account` to client.

### PR DESCRIPTION
**Pull request by cyberhoward**
Adds a function to the AbstractClient to get the last account that a signer created.

### Checklist

- [ ] CI is green.
- [ ] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-client/src/client.rs [hunk 1]
```diff
@@ -119,3 +120,56 @@ impl<Chain: MutCwEnv> AbstractClient<Chain> {
...
+            let addresses = self.environment().state().get_all_addresses()?;
+
+            // Now search for all the keys that start with "abstract:manager-x" and return the one which has the highest x.
+            let mut last_account: Option<(u32, Account<Daemon>)> = None;
+            for (id, _) in addresses {
+                if !id.starts_with(MANAGER) {
+                    continue;
+                }
+
+                let account_id_str = id.rsplitn(2, '-').next().unwrap();
```
Comment (Buckram123) [inline framework/packages/abstract-client/src/client.rs:148]: Wouldn't it take only sequence here, for example
```
"abstract:manager-local-5".rsplitn(2, '-').next().unwrap() == "5" 
```
I would prefer to use `split_once` here

File: framework/packages/abstract-core/src/objects/account/account_id.rs [hunk 2]
```diff
@@ -76,6 +76,43 @@ impl AccountId {
     }
 }
 
+impl TryFrom<&str> for AccountId {
+    type Error = AbstractError;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let mut val_iter = value.rsplitn(2, '-');
```
Comment (Buckram123) [inline framework/packages/abstract-core/src/objects/account/account_id.rs:83]: let's use rsplit_once instead

File: framework/packages/abstract-client/src/client.rs [hunk 3]
```diff
@@ -119,3 +120,56 @@ impl<Chain: MutCwEnv> AbstractClient<Chain> {
...
+        /// Retrieve the last account created by the client.
+        /// Returns `None` if no account has been created yet.
+        pub fn get_last_account(&self) -> AbstractClientResult<Option<Account<Daemon>>> {
+            let addresses = self.environment().state().get_all_addresses()?;
+
+            // Now search for all the keys that start with "abstract:manager-x" and return the one which has the highest x.
+            let mut last_account: Option<(u32, Account<Daemon>)> = None;
+            for (id, _) in addresses {
+                if !id.starts_with(MANAGER) {
+                    continue;
```
Comment (Buckram123) [inline framework/packages/abstract-client/src/client.rs:145]: Actually, can you cut it into different method, that takes `HashMap<String, Addr>` so it's unit-testable?

**Review by Buckram123 — CHANGES_REQUESTED** [review]
Looks like a small bug, but maybe I'm wrong if it's tested

File: framework/packages/abstract-client/src/client.rs (hunk 1 above)
Comment (CyberHoward) [reply framework/packages/abstract-client/src/client.rs:148]: Ah yeah you're right!

File: framework/packages/abstract-client/src/client.rs (hunk 3 above)
Comment (CyberHoward) [reply framework/packages/abstract-client/src/client.rs:145]: yes :)

File: framework/packages/abstract-core/src/objects/account/account_id.rs (hunk 2 above)
Comment (CyberHoward) [reply framework/packages/abstract-core/src/objects/account/account_id.rs:83]: Here's me thinking I know the `std` crate xd

File: framework/packages/abstract-client/src/client.rs (hunk 3 above)
Comment (CyberHoward) [reply framework/packages/abstract-client/src/client.rs:145]: Actually this will be hard to test because we need an actual Daemon object with an owner and we can't easily test all the different edge cases.

Comment (Buckram123) [reply framework/packages/abstract-client/src/client.rs]: > Actually this will be hard to test because we need an actual Daemon object with an owner and we can't easily test all the different edge cases.

Hmm, is it tho? Can't that helper method can take `CwEnv` instead and test against Mock?

File: framework/packages/abstract-client/src/client.rs [hunk 4]
```diff
@@ -102,6 +105,34 @@ impl<Chain: CwEnv> AbstractClient<Chain> {
             .map_err(Into::into)
             .map_err(Into::into)
     }
+
+    // Retrieve the last account created by the client.
+    /// Returns `None` if no account has been created yet.
+    pub fn get_last_account(&self) -> AbstractClientResult<Option<Account<Chain>>> {
```
Comment (Buckram123) [inline framework/packages/abstract-client/src/client.rs:111]: Woud be nice to mention that it would only load accounts stored in this system

**Review by Buckram123 — APPROVED** [review]
LGTM!
What do you think about method that will fetch all account ids from the chain owned by the sender? Or would you consider it only something for the FE?

**Comment by CyberHoward** [comment]
> LGTM! What do you think about method that will fetch all account ids from the chain owned by the sender? Or would you consider it only something for the FE?

That would be very nice to have! But we can add that in a different PR. :)
https://linear.app/abstract-sdk/issue/ABS-287/implement-an-account-fetcher-for-abstract-client

File: framework/packages/abstract-client/src/client.rs (hunk 4 above)
Comment (CyberHoward) [reply framework/packages/abstract-client/src/client.rs]: done!
